### PR TITLE
Jaxws-2.3 feature needs IBM-API-Package header updated

### PR DESCRIPTION
**Pull request by neuwerk**
This PR has been expanded to included a whole list of fixes that were needed.

### CODE REVIEW

**Comment by neuwerk** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.internal.jaxws-2.3.feature [hunk 1]
```diff
@@ -33,7 +34,7 @@ Subsystem-Name: Internal Java Web Services 2.3
  com.ibm.websphere.javaee.jws.1.0; require-java:="9"; location:="dev/api/spec/,lib/"; mavenCoordinates="javax.jws:jsr181-api:1.0-MR1",\
  com.ibm.websphere.javaee.jaxws.2.3; location:="dev/api/spec/"; apiJar=false,\
  com.ibm.ws.jaxws.tools.2.2.10, \
- com.ibm.ws.jaxws.2.3.common, \
+ com.ibm.ws.jaxws.2.3.common; start-phase:=CONTAINER_LATE, \
```
Comment (neuwerk) [inline dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.internal.jaxws-2.3.feature]: Copied a change from jaxws-2.2 since I'm assuming the same start-up issues will occur with this if not included.

File: dev/com.ibm.websphere.appserver.features/visibility/public/jaxws-2.3/com.ibm.websphere.appserver.jaxws-2.3.feature [hunk 2]
```diff
@@ -5,6 +5,29 @@ singleton=true
 IBM-App-ForceRestart: uninstall, \
  install
 IBM-ShortName: jaxws-2.3
+IBM-API-Package: \
```
Comment (neuwerk) [inline dev/com.ibm.websphere.appserver.features/visibility/public/jaxws-2.3/com.ibm.websphere.appserver.jaxws-2.3.feature]: Had to add API-Packages as none of these were being exposed to the application class path

File: dev/com.ibm.ws.jaxws.2.3.common/bnd.bnd [hunk 3]
```diff
@@ -53,7 +53,6 @@ Import-Package: \
 
 Private-Package:\
    com.ibm.ws.jaxws.*, \
-   com.ibm.ws.jaxws23.component, \
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.3.common/bnd.bnd]: This package doesn't exist, I added it to test a fix but removed it when it had no effect

File: dev/com.ibm.ws.jaxws.2.3.common/bnd.bnd [hunk 4]
```diff
@@ -167,11 +166,6 @@ Service-Component: \
      implementation:=com.ibm.ws.jaxws.catalog.OASISCatalogApplicationBusListener; \
      provide:='com.ibm.ws.jaxws.bus.LibertyApplicationBusListener'; \
      properties:="service.vendor=IBM", \
-  com.ibm.ws.jaxws.http.transport.factory.provider; \
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.3.common/bnd.bnd]: This was a big source of the problems with jaxws-2.3. Because off the way that jaxws-2.2 used the integration layer to hook-up to the CXF HTTP Transport module, that same integration layer no longer works with jaxws-2.3 and CXF 3.3. So I've removed our Liberty specific extensions of the CXF transports http module, this includes this Service-Component.

File: dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/client/LibertyProviderImpl.java [hunk 5]
```diff
@@ -53,13 +55,31 @@
     @Override
     public ServiceDelegate createServiceDelegate(URL url, QName qname,
                                                  @SuppressWarnings("rawtypes") Class cls) {
+        
+        //Eager initialize the StaxUtils
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/client/LibertyProviderImpl.java]: Needed to update the client-side integration to allow CXF to use the STaX from the RI since that's what we ship with OL. We need to change this to use WoodStox in the future.

File: dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/client/injection/ServiceRefObjectFactory.java [hunk 6]
```diff
@@ -112,7 +111,6 @@ protected void deActivate(ComponentContext cCtx) {
     protected void setSecurityConfigurationService(ServiceReference<JaxWsSecurityConfigurationService> serviceRef) {
         securityConfigSR.setReference(serviceRef);
         LibertyProviderImpl.setSecurityConfigService(securityConfigSR);
-        LibertyHTTPTransportFactory.setSecurityConfigService(securityConfigSR);
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/client/injection/ServiceRefObjectFactory.java]: See the above comment about removing Liberty's extensions of CXF's Transport HTTP Module

File: dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/support/JaxWsServiceImpl.java [hunk 7]
```diff
@@ -51,17 +51,24 @@ protected void activate(ComponentContext cc) {
...
         }
 
-        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-            Tr.debug(tc, "Insecure Stax property is set to: " + System.getProperty(StaxUtils.ALLOW_INSECURE_PARSER));
-        }
-
         //Eager initialize the StaxUtils
         try {
             THREAD_CONTEXT_ACCESSOR.setContextClassLoader(Thread.currentThread(), StAXUtils.getStAXProviderClassLoader());
+            if(System.getProperty(StaxUtils.ALLOW_INSECURE_PARSER) == null) {
```
Comment (neuwerk) [inline dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/support/JaxWsServiceImpl.java]: See above comment about STaX, this is for the Service side

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/bnd.bnd [hunk 8]
```diff
@@ -28,4 +28,5 @@ instrument.classesExcludes: org/apache/cxf/common/jaxb/*.class
   com.ibm.ws.org.osgi.annotation.versioning;version=latest,\
   com.ibm.websphere.org.osgi.service.component;version=latest,\
   com.ibm.wsspi.org.osgi.service.component.annotations;version=latest,\
+  com.ibm.websphere.org.osgi.service.cm;version=latest,\
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/bnd.bnd]: Pulling in the CXF Activator as the cxf.core.3.2's Bundle-Activator (as you can see below) required another osgi dependency to compile

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/bus/osgi/CXFActivator.java [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+                                                                     ManagedServiceFactory.class,
+                                                                     workQueues,
+                                                                     ManagedWorkQueueList.FACTORY_PID);
+
+        extensions = new ArrayList<>();
+        extensions.add(createOsgiBusListenerExtension(context));
+        extensions.add(createManagedWorkQueueListExtension(workQueues));
+        ExtensionRegistry.addExtensions(extensions);
+
+        // Prevent the CXFActivator from loading Blueprint
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/bus/osgi/CXFActivator.java]: Switched from using the NoOpActivator to the CXFActivator but with the provision that Blueprint used in our version.

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.bindings.soap.3.2/bnd.overrides [hunk 10]
```diff
@@ -14,29 +14,15 @@ cxfVersion=3.3.3
 
 Bundle-SymbolicName: com.ibm.ws.org.apache.cxf.cxf.rt.bindings.soap.3.2
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.bindings.soap.3.2/bnd.overrides]: I had to remove all of the NoOpActivators in all of the jaxws-2.3 specifc CXF bundles because the CXF activators are responsible for loading Bus extensions.

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.bindings.soap.3.2/src/org/apache/cxf/bindings/soap/SoapBinding.java [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+    public Message createMessage() {
+        SoapMessage soapMessage = new SoapMessage(version);
+        soapMessage.put(Message.CONTENT_TYPE, soapMessage.getVersion().getContentType());
+        return soapMessage;
+    }
+
+    public Message createMessage(Message m) {
+       
+        SoapMessage soapMessage = new SoapMessage(m);
+        if (m.getExchange() != null) {
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.bindings.soap.3.2/src/org/apache/cxf/bindings/soap/SoapBinding.java]: Made some changes to handle how the Message object has been updated in OL

File: dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/catalog/OASISCatalogApplicationBusListener.java [hunk 12]
```diff
@@ -41,7 +41,7 @@ public void initComplete(Bus bus) {
         if (bus == null) {
             return;
         }
-
+
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/catalog/OASISCatalogApplicationBusListener.java]: weird whitespace is the only change to this file

File: dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/support/LibertyJaxWsServerFactoryBean.java [hunk 13]
```diff
@@ -72,4 +75,5 @@ public LibertyJaxWsServerFactoryBean(JaxWsServiceFactoryBean serviceFactoryBean)
         }
         return HandlerChainInfoBuilder.sortHandlers(handlers);
     }
+
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/support/LibertyJaxWsServerFactoryBean.java]: unused imports and new whitespace?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/http/URLConnectionHTTPConduit.java [hunk 14]
```diff
@@ -125,6 +125,16 @@ private HttpURLConnection createConnection(Message message, Address address, HTT
         return connectionFactory.createConnection(clientParameters,
                                                   proxy != null ? proxy : address.getDefaultProxy(), url);
     }
+    
+    
+    public void setAddress(String address) throws IOException {
+        try {
+        defaultAddress = new Address(address);
+        
+    } catch (URISyntaxException e) {
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/org/apache/cxf/transport/http/URLConnectionHTTPConduit.java]: catch block indentation

**Comment by neuwerk** [comment]
#build

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.management.3.2/resources/.gitignore [hunk 15]
```diff
@@ -0,0 +1,6 @@
+*Messages.nlsprops
```
Comment (BerksanAtes) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.management.3.2/resources/.gitignore]: Do we need to ignore these nlsprops?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.addr.3.2/resources/.gitignore [hunk 16]
```diff
@@ -0,0 +1,6 @@
+*Messages.nlsprops
+*APIMessages.nlsprops
+*version.nlsprops
+*Message.nlsprops
```
Comment (BerksanAtes) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.ws.addr.3.2/resources/.gitignore]: Somehow .gitignore is adding *.nlsprops to be ignored

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.wsdl.3.2/resources/.gitignore [hunk 17]
```diff
@@ -0,0 +1,6 @@
+*Messages.nlsprops
+*APIMessages.nlsprops
+*version.nlsprops
+*Message.nlsprops
```
Comment (BerksanAtes) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.wsdl.3.2/resources/.gitignore]: Do we need to ignore *.nlsprops files?

**Comment by neuwerk** [comment]
#8716

**Comment by neuwerk** [comment]
#build

**Comment by neuwerk** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.internal.jaxws-2.3.feature (hunk 1 above)
Comment (WhiteCat22) [reply dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.internal.jaxws-2.3.feature]: good bet

File: dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/StringMapImpl.java [hunk 18]
```diff
@@ -54,6 +54,7 @@ public StringMapImpl(Map<String, Object> i) {
         put(key.getName(), value);
     }
 
+
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/StringMapImpl.java]: whitespace!

**Comment by neuwerk** [comment]
@tevans78 thank you for the reminder to keep my PRs clean 😄  I've squashed it down to the one commit.

**Comment by neuwerk** [comment]
#build

**Comment by neuwerk** [comment]
#build

**Comment by neuwerk** [comment]
Failure in PB is unrelated to my changes, as it's a known failure.

**Comment by neuwerk** [comment]
#run-libby-bot

**Comment by neuwerk** [comment]
#build

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/com/ibm/ws/cxf/transport/http/NoOpActivator.java [hunk 19]
```diff
@@ -1,14 +1,5 @@
...
- * All rights reserved. This program and the accompanying materials
- * are made available under the terms of the Eclipse Public License v1.0
- * which accompanies this distribution, and is available at
- * http://www.eclipse.org/legal/epl-v10.html
- *
- * Contributors:
- *     IBM Corporation - initial API and implementation
- *******************************************************************************/
-package com.ibm.ws.jaxws23.cxf.rt.wsdl;
+package com.ibm.ws.cxf.transport.http;
```
Comment (tevans78) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/com/ibm/ws/cxf/transport/http/NoOpActivator.java]: Why did you remove the copyright header?

Comment (neuwerk) [reply dev/com.ibm.ws.org.apache.cxf.cxf.rt.transports.http.3.2/src/com/ibm/ws/cxf/transport/http/NoOpActivator.java]: That was not intentional! Just a bad copy paste. Thanks for catching it!

**Comment by neuwerk** [comment]
#run-libby-bot

**Comment by neuwerk** [comment]
#build

**Comment by neuwerk** [comment]
#run-libby-bot
